typing/rec_check.ml

```diff
@@ -163,8 +163,24 @@ let classify_expression : Typedtree.expression -> sd =
 
     | Texp_construct (_, {cstr_tag = Cstr_unboxed}, [e]) ->
         classify_expression env e
-    | Texp_construct _ ->
-        Static
+    | Texp_construct (_, _, exprs) ->
+        let is_constant =
+          List.fold_left (fun is_constant expr ->
+              is_constant &&
+              match classify_expression env expr with
+              | Constant -> true
+              | _ -> false)
+            true exprs
```

Comment:
This should be `List.for_all`.

Reply:
I would suggest defining a `is_constant : expr -> bool` helper function (probably as a local binding below `let rec classify_expression ... =`?) to use here and below as well.